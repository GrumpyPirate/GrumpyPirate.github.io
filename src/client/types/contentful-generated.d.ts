// THIS FILE IS AUTOMATICALLY GENERATED. DO NOT MODIFY IT.

import { Asset, Entry } from 'contentful';
import { Document } from '@contentful/rich-text-types';

export interface IAboutPageSectionFields {
  /** Title */
  title: string;

  /** slug */
  slug: string;

  /** Icons */
  icons?: Record<string, any> | undefined;

  /** Content */
  content: string;

  /** order */
  order: number;
}

/** An individual section for the about page */
export interface IAboutPageSection extends Entry<IAboutPageSectionFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'aboutPageSection';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}

export interface IPortfolioItemFields {
  /** title */
  title: string;

  /** url */
  url?: string | undefined;

  /** tech */
  tech?: Record<string, any> | undefined;

  /** Supporting Image */
  supportingImage?: Asset | undefined;

  /** Header Image */
  headerImage: Asset;

  /** Device Previews */
  devicePreviews?: (Asset)[] | undefined;

  /** slug */
  slug: string;

  /** Description */
  description: string;

  /** description-short */
  descriptionShort: string;
}

export interface IPortfolioItem extends Entry<IPortfolioItemFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'portfolioItem';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}

type CONTENT_TYPE = 'aboutPageSection' | 'portfolioItem';

type LOCALE_CODE = 'en-GB';

type CONTENTFUL_DEFAULT_LOCALE_CODE = 'en-GB';
