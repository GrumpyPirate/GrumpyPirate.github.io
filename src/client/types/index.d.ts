import { ReactNode } from 'react';

interface ChildrenProps {
  children: ReactNode;
}

interface AboutSectionIcon {
  name: string;
  large?: boolean;
}

interface AboutSectionFormatted {
  id: string;
  title: string;
  icons?: AboutSectionIcon[];
  slug: string;
  content: string;
  order: number;
}

interface PortfolioItemTech {
  icon: string;
  name: string;
}

interface PortfolioItemFormatted {
  id: string;
  slug: string;
  title: string;
  descriptionShort: string;
  description: string;
  url: string;
  tech: PortfolioItemTech[];
  supportingImageSrc: string;
  headerImgSrc: string;
  previews: {
    desktop: string;
    tablet: string;
    mobile: string;
  };
  createdAt: string;
}

export {
  AboutSectionIcon,
  AboutSectionFormatted,
  ChildrenProps,
  PortfolioItemTech,
  PortfolioItemFormatted,
};
