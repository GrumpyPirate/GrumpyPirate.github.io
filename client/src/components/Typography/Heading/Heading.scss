@import 'core';

.heading {
  font-family: $headings-font-family;
  font-weight: $headings-font-weight;
  letter-spacing: $headings-letter-spacing;
  line-height: $headings-line-height;
  margin: 0 0 $headings-margin-bottom;
  text-transform: uppercase;

  // Modifiers
  &--level {
    &-1 {
      font-size: $font-size-h1-mobile;

      @include respond-up('md') {
        font-size: $font-size-h1;
      }
    }

    &-2 {
      font-size: $font-size-h2-mobile;

      @include respond-up('md') {
        font-size: $font-size-h2;
      }
    }

    &-3 {
      font-size: $font-size-h3-mobile;

      @include respond-up('md') {
        font-size: $font-size-h3;
      }
    }

    &-4 {
      font-size: $font-size-h4-mobile;

      @include respond-up('md') {
        font-size: $font-size-h4;
      }
    }

    &-5 {
      font-size: $font-size-h5-mobile;

      @include respond-up('md') {
        font-size: $font-size-h5;
      }
    }

    &-6 {
      font-size: $font-size-h6-mobile;

      @include respond-up('md') {
        font-size: $font-size-h6;
      }
    }
  }
}
