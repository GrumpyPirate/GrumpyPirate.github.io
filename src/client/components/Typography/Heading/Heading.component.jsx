import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import classes from './Heading.component.scss';

const Heading = ({ level, displayLevel, text, className }) => {
  const levelToDisplay = displayLevel || level;
  const classNames = classnames(classes['heading'], classes[`heading--level-${levelToDisplay}`], {
    [className]: className,
  });

  switch (level) {
    case 1:
      return <h1 className={classNames}>{text}</h1>;
    case 2:
      return <h2 className={classNames}>{text}</h2>;
    case 3:
      return <h3 className={classNames}>{text}</h3>;
    case 4:
      return <h4 className={classNames}>{text}</h4>;
    case 5:
      return <h5 className={classNames}>{text}</h5>;
    case 6:
      return <h6 className={classNames}>{text}</h6>;
    default:
      return <h5 className={classNames}>{text}</h5>;
  }
};

Heading.propTypes = {
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  displayLevel: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Heading.defaultProps = {
  level: 5,
  displayLevel: null,
  className: null,
};

export default Heading;
