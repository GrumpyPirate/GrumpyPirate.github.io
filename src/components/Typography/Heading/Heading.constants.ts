import { css } from 'styled-components';

import {
  fontSizeH1,
  fontSizeH1Mobile,
  fontSizeH2,
  fontSizeH2Mobile,
  fontSizeH3,
  fontSizeH3Mobile,
  fontSizeH4,
  fontSizeH4Mobile,
  fontSizeH5,
  fontSizeH5Mobile,
  fontSizeH6,
  fontSizeH6Mobile,
  mediaQueries,
  rem,
} from 'styles';

export const heading1Styles = css`
  font-size: ${rem(fontSizeH1Mobile)};

  @media ${mediaQueries.md} {
    font-size: ${rem(fontSizeH1)};
  }
`;
export const heading2Styles = css`
  font-size: ${rem(fontSizeH2Mobile)};

  @media ${mediaQueries.md} {
    font-size: ${rem(fontSizeH2)};
  }
`;
export const heading3Styles = css`
  font-size: ${rem(fontSizeH3Mobile)};

  @media ${mediaQueries.md} {
    font-size: ${rem(fontSizeH3)};
  }
`;
export const heading4Styles = css`
  font-size: ${rem(fontSizeH4Mobile)};

  @media ${mediaQueries.md} {
    font-size: ${rem(fontSizeH4)};
  }
`;
export const heading5Styles = css`
  font-size: ${rem(fontSizeH5Mobile)};

  @media ${mediaQueries.md} {
    font-size: ${rem(fontSizeH5)};
  }
`;
export const heading6Styles = css`
  font-size: ${rem(fontSizeH6Mobile)};

  @media ${mediaQueries.md} {
    font-size: ${rem(fontSizeH6)};
  }
`;
