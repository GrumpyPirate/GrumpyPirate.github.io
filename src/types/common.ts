import { Action, ThunkAction } from '@reduxjs/toolkit';

export interface ClassNameProps {
  className?: string;
}

export type IconGlyph =
  | 'airplay'
  | 'angular'
  | 'babel'
  | 'bootstrap'
  | 'bulma'
  | 'chevron-down'
  | 'chevron-left'
  | 'contentful'
  | 'divio'
  | 'django'
  | 'foundation'
  | 'grunt'
  | 'gulp'
  | 'hash'
  | 'heart'
  | 'html5'
  | 'illustrator'
  | 'image'
  | 'invision'
  | 'javascript'
  | 'laravel'
  | 'loader'
  | 'nextjs'
  | 'photoshop'
  | 'postcss'
  | 'react'
  | 'reactivex'
  | 'redux'
  | 'sass'
  | 'shopify'
  | 'sketch'
  | 'storybook'
  | 'styled-components'
  | 'typescript'
  | 'webpack';

export interface AboutSectionIcon {
  name: IconGlyph;
  large?: boolean;
}

export interface AboutSectionFormatted {
  id: string;
  title: string;
  icons?: AboutSectionIcon[];
  slug: string;
  content: string;
  order: number;
}

export interface PortfolioItemTech {
  icon: IconGlyph;
  name: string;
}

export interface PortfolioItemFormatted {
  id: string;
  slug: string;
  title: string;
  descriptionShort: string;
  description: string;
  url: string;
  tech: PortfolioItemTech[];
  supportingImageSrc: string;
  headerImgSrc: string;
  previews: {
    desktop: string;
    tablet: string;
    mobile: string;
  };
  createdAt: string;
}

export type AppThunk = ThunkAction<void, any, unknown, Action<string>>;
export interface GenericActionError {
  message?: string;
}
