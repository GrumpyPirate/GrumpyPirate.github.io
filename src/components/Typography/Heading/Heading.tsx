import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import {
  headingsFontFamily,
  headingsFontWeight,
  headingsLetterSpacing,
  headingsLineHeight,
  headingsMarginBottom,
} from 'styles';

import {
  heading1Styles,
  heading2Styles,
  heading3Styles,
  heading4Styles,
  heading5Styles,
  heading6Styles,
} from './Heading.constants';
import { HeadingProps } from './Heading.types';

const Heading: FunctionComponent<HeadingProps> = ({ className, text, level = 5 }) => {
  switch (level) {
    case 1:
      return <h1 className={className}>{text}</h1>;
    case 2:
      return <h2 className={className}>{text}</h2>;
    case 3:
      return <h3 className={className}>{text}</h3>;
    case 4:
      return <h4 className={className}>{text}</h4>;
    case 6:
      return <h6 className={className}>{text}</h6>;
    case 5:
    default:
      return <h5 className={className}>{text}</h5>;
  }
};

export default styled(Heading)`
  font-family: ${headingsFontFamily};
  font-weight: ${headingsFontWeight};
  letter-spacing: ${headingsLetterSpacing};
  line-height: ${headingsLineHeight};
  margin: 0 0 ${headingsMarginBottom};
  text-transform: uppercase;

  ${({ displayLevel, level }) => {
    const levelToDisplay = displayLevel || level;

    switch (levelToDisplay) {
      case 1:
        return heading1Styles;
      case 2:
        return heading2Styles;
      case 3:
        return heading3Styles;
      case 4:
        return heading4Styles;
      case 6:
        return heading6Styles;
      case 5:
      default:
        return heading5Styles;
    }
  }}
`;
