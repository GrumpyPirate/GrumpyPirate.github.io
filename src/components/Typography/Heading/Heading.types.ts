import { PropsWithClassName } from 'types/common';

export type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;

export type HeadingProps = PropsWithClassName<{
  className?: string;
  displayLevel?: HeadingLevel;
  level?: HeadingLevel;
  text: string;
}>;
