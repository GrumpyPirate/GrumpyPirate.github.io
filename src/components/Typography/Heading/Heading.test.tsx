import { render } from '@testing-library/react';
import React from 'react';

import Heading from './Heading';
import fixture from './Heading.fixture';
import { HeadingLevel } from './Heading.types';

describe('Components', () => {
  describe('Heading', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const { container } = render(<Heading {...props} />);

      expect(container.firstChild).toMatchSnapshot();
    });

    describe('props: level', () => {
      it('should render an h5 element, by default', () => {
        const { container } = render(<Heading {...props} />);

        expect(container.querySelectorAll('h5')).toHaveLength(1);
      });

      it('should render appropriate heading elements, when given level props of 1-6', () => {
        ([1, 2, 3, 4, 5, 6] as HeadingLevel[]).forEach((level) => {
          const { container } = render(<Heading {...{ ...props, level }} />);

          expect(container.querySelectorAll(`h${level}`)).toHaveLength(1);
        });
      });
    });
  });
});
