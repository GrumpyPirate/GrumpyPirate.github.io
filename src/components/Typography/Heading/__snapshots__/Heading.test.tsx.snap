// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Heading should render as expected, without crashing 1`] = `
.c0 {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
}

@media (min-width:48em) {
  .c0 {
    font-size: 1rem;
  }
}

<h5
  class="c0"
>
  Facilis dicta totam assumenda
</h5>
`;
